src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketsTelemetry.cs

```diff
@@ -82,68 +75,89 @@ public void AcceptStop()
         }
 
         [Event(7, Level = EventLevel.Error)]
-        public void AcceptFailed(SocketError error, string? exceptionMessage)
+        private void AcceptFailed(SocketError error, string? exceptionMessage)
         {
             if (IsEnabled(EventLevel.Error, EventKeywords.All))
             {
-                WriteEvent(eventId: 7, (int)error, exceptionMessage ?? string.Empty);
+                WriteEvent(eventId: 7, (int)error, exceptionMessage);
             }
         }
 
         [NonEvent]
         public void ConnectStart(Internals.SocketAddress address)
         {
-            ConnectStart(address.ToString());
+            if (IsEnabled(EventLevel.Informational, EventKeywords.All))
+            {
+                ConnectStart(address.ToString());
+            }
         }
 
         [NonEvent]
-        public void ConnectStart(EndPoint address)
+        public void AfterConnect(SocketError error, string? exceptionMessage = null)
         {
-            ConnectStart(address.ToString());
-        }
+            if (error == SocketError.Success)
+            {
+                Debug.Assert(exceptionMessage is null);
+                Interlocked.Increment(ref _outgoingConnectionsEstablished);
+            }
+            else
+            {
+                if (error == SocketError.OperationAborted || error == SocketError.NotSocket)
+                {
+                    ConnectCanceled();
+                }
 
-        [NonEvent]
-        public void ConnectCanceledAndStop()
-        {
-            ConnectCanceled();
-            ConnectStop();
-        }
+                ConnectFailed(error, exceptionMessage);
+            }
 
```

Comment:
It's intentional that if a connect is canceled, we do ConnectCanceled, ConnectFailed, and ConnectStop?

Reply:
Yes, as ConnectCanceled and ConnectFailed are logged with different severity levels.

Reply:
The levels here seem a little backward.  If I'm trying to reduce logging noise by only listening to errors and nothing more verbose, I'll end up getting a failed event, even though the operation was actually canceled (which likely makes the failure less important to pay attention to).

Reply:
I kept the levels as introduced in #38620, but you are correct, they don't make sense like this - the severities should be swapped.

Reply:
I'm not sure they should be swapped, either: Canceled isn't more serious than Failed.  But the result ends up being weird.

Reply:
Thinking about this more, considering the relatively low frequency of these events (one per socket connect), I don't see a lot of value in having a separate severity level just for ConnectCanceled. And if they are at the same level, they are redundant as Canceled is a strict subset of Failed and doesn't include any extra useful info.

Reply:
> If we always log ConnectFailed even when it was canceled, why do we need the canceled event at all? We expect folks will be doing something specific with that cancellation event

With the current Levels, the canceled event has very little/no value.
Would the move here be to keep it as is or remove the canceled event outright?

Reply:
I'd suggest deleting it until there's a demonstrated need. @geoffkizer?

Does this pattern show up for operations other than Connect? There's of course something to be said for consistency.

Reply:
> Does this pattern show up for operations other than Connect?

No, Accept doesn't have a corresponding canceled event.

Net.Http, Net.NameResolution and Net.Security all have `Start`/`Failed`/`Stop` events.

Reply:
Removed `ConnectCanceled`